src/parse-cookies.ts

```diff
@@ -1,13 +1,39 @@
 const parseCookies = (s: string): { [key: string]: string } => {
   if (s.length === 0) return {};
   const parsed: { [key: string]: string } = {};
-  const pattern = new RegExp("\\s*;\\s*");
-  s.split(pattern).forEach((i) => {
-    const [encodedKey, encodedValue] = i.split("=");
-    const key = decodeURIComponent(encodedKey);
-    const value = decodeURIComponent(encodedValue);
-    parsed[key] = value;
-  });
+  const pairs = s.split(';')
+
+  for (let i = 0; i < pairs.length; i++) {
+    const pair = pairs[i];
+    const index = pair.indexOf('=')
+
+    // skip things that don't look like key=value
+    if (index < 0) {
+      continue;
+    }
+
+    const key = pair.substring(0, index).trim()
+
+    // only assign once
+    if (undefined == parsed[key]) {
```

Comment:
I'm curious why you chose to have `undefined` to be on the left hand side of this statement. Also are you sure `==` is the correct operator?

![image](https://user-images.githubusercontent.com/7682123/211066187-bfffdcd0-2067-4273-a9fd-4eebb8bdae9f.png)

You could use `===` instead. This would ensure that your parsed key is actually `undefined`